tests/EmailTest.elm

```diff
@@ -18,36 +18,130 @@ suite =
                     parse "helloworld.com"
                         |> Expect.err
             ]
-        , describe "isValid"
-            [ test "valid email" <|
-                \_ ->
-                    isValid "hello@world.com"
-                        |> Expect.equal True
-            , test "invalid char in local part" <|
-                \_ ->
-                    isValid "he^llo@world.com"
-                        |> Expect.equal False
-            , test "invalid char in domain part" <|
-                \_ ->
-                    isValid "hello@worl^d.com"
-                        |> Expect.equal False
-            , test "bad domain" <|
-                \_ ->
-                    isValid "hello@world"
-                        |> Expect.equal False
-            , test "missing domain" <|
-                \_ ->
-                    isValid "hello@"
-                        |> Expect.equal False
-            , test "missing local" <|
-                \_ ->
-                    isValid "@world.com"
-                        |> Expect.equal False
-            ]
+        , describe "valid emails" <| validEmailTests
+        , describe "invalid emails" <| invalidEmailTests
```

Comment:
This code organization is nice in the tests. Thanks!

Reply:
Glad you like it! Thanks for merging :)